src/compiler/crystal/macros/methods.cr

```diff
@@ -2668,14 +2668,14 @@ private def visibility_to_symbol(visibility)
   Crystal::SymbolLiteral.new(visibility_name)
 end
 
-private def macro_raise(node, args, interpreter)
+private def macro_raise(node, args, interpreter, exception_type)
   msg = args.map do |arg|
     arg.accept interpreter
     interpreter.last.to_macro_id
   end
   msg = msg.join " "
 
-  node.raise msg, exception_type: Crystal::MacroRaiseException
+  node.raise msg, exception_type: exception_type
```

Comment:
Allows differentiating a node specific versus top level raise to make the logic simpler as each case is slightly different in how it should be ordered.